### ISSUE TITLE
Feature Request: Concurrency

### ISSUE THREAD

**Issue by tcullum-gpsw** [dxa4481/truffleHog#126]
dxa4481, I saw your talk and am aware that you are looking to implement multithreading. I'm also interested in some form of concurrency, but after reading about the model used in Python, I'm wondering if it would be worth it? Apparently python does not have "true" multithreading, however, perhaps we could utilize multiprocessing? Do you have experience in this area? Thought about doing a PR. Thanks.

**Comment by dxa4481** [comment]
Yeah so the major issues I've had here is OOM issues. I'm not sure the best way to navigate memory management issues with threads, potentially each thread may load in files multiple megabytes big, and run complex regex on them, if there's some good python resource for managing threads in python and memory, in a way that can suspend or pause a thread when the system reaches OOM, that'd be where I'd want to start.

**Comment by shoaibali123** [comment]
Hi dxa4481, it doesn't work for me. I see the script in master branch but when I used it didn't work for me. Can you give more details? Thanks.

**Comment by dustin-decker** [comment]
Hey there, we've just released the next major version of TruffleHog! 

It includes concurrency!

It is a complete rewrite that scans more data sources and now supports detecting and verifying over 600 credentials.
Please check it out when you can.

https://trufflesecurity.com/blog/introducing-trufflehog-v3

We're no longer maintaining v2 so I am closing this issue.
